asab/library/providers/azurestorage.py

```diff
@@ -128,8 +145,54 @@ async def list(self, path: str) -> list:
 
 		return items
 
+	def load_from_cache(self):
+		"""
+		Load the lookup data (bytes) from cache.
+		"""
+		if self.UseCache is False:
+			return False
+		# Load the ETag from cached file, if have one
+		if not os.path.isfile(self.CachePath):
+			L.warning("Cache '{}': not a file".format(self.CachePath))
+			return False
+
+		if not os.access(self.CachePath, os.R_OK):
+			L.warning("Cannot read cache from '{}'".format(self.CachePath))
+			return False
+
+		try:
+			with open(self.CachePath, 'rb') as f:
+				tlen, = struct.unpack(r"<L", f.read(struct.calcsize(r"<L")))
+				etag_b = f.read(tlen)
+				self.ETag = etag_b.decode('utf-8')
+				f.read(1)
+				data = f.read()
+			return data
+		except Exception as e:
+			L.warning("Failed to read content of lookup cache '{}' from '{}': {}".format(self.Id, self.CachePath, e))
+			os.unlink(self.CachePath)
+		return False
+
+	def save_to_cache(self, data):
+		if self.UseCache is False:
+			return
+		dirname = os.path.dirname(self.CachePath)
+		if not os.path.isdir(dirname):
+			os.makedirs(dirname)
+
+		with open(self.CachePath, 'wb') as fo:
+			# Write E-Tag and '\n'
+			etag_b = self.ETag.encode('utf-8')
+			fo.write(struct.pack(r"<L", len(etag_b)) + etag_b + b'\n')
```

Comment:
You need to store ETag in the dedicated file, with e.g. `.etag` extension (added to the `self.CachePath`)